main/screen.c

```diff
@@ -417,7 +489,19 @@ static void screen_update_cb(lv_timer_t * timer)
 
 void screen_next()
 {
-    screen_show(current_screen == SCR_CAROUSEL_END ? SCR_CAROUSEL_START : current_screen + 1);
+    screen_t next_scr = current_screen;
+
+    // Loop to find the next screen that should be displayed
+    do {
+        next_scr++; // Advance to the next screen candidate
+        if (next_scr > SCR_CAROUSEL_END) { // If past the end of carousel
+            next_scr = SCR_CAROUSEL_START; // Wrap around to the start of carousel
+        }
+        // If the candidate screen is SCR_WIFI_RSSI AND this is NOT a bigger display,
+        // then this screen should be skipped, and the loop will continue to find the next one.
+    } while (next_scr == SCR_WIFI_RSSI && GLOBAL_STATE->DISPLAY_CONFIG.v_res / 8 != 8);
```

Comment:
We have to think about this a bit more. For now I would revert this bit, and just add the new screen for all displays. If we want to have different screen designs for different resolutions, it would make sense to do that in `screen_start`, only once.

Reply:
F.e. we could make 2 separate designs for the Wi-Fi, and either add one or the other on the SCR_WIFI_RSSI position. Or have a completely dynamic bit of the carousel, that a small screen has 4 screens (as an example), but a larger screen only has 3, with different designs.